### PR TITLE
Support recursion in the CODEOWNERS file.

### PR DESCRIPTION
**Pull request by shaleh**
Uses pattern.match and os.path.

### CODE REVIEW

**Comment by nmusolino** [comment]
Thank you, ashaleh, for your MR. I apologize for not responding earlier.

Would you be able to update the test for this? There is a [test](https://github.com/nmusolino/codeowners/blob/e46c0da7f1d86d73fd1e5492096847a53ff7059c/tests/test_codeowners.py#L66) currently marked `xfail`. Does that test pass after your change?

I would normally be able to check myself, but I have limited access to my dev machine during a Coronavirus lockdown.

**Comment by shaleh** [comment]
@nmusolino I will give it a look. Thanks for a pointer to more tests.

**Comment by shaleh** [comment]
Yay for inconsistent Python docs.

`match` on pathlib does not recurse. `glob` does. So I can make this work with real files on a real disk but it would not support simply asking "does this string this glob". Fun.

**Comment by shaleh** [comment]
I will ponder this and get back to you.

**Comment by shaleh** [comment]
@nmusolino `fnmatch` does not support recursion. `glob` does, but only with real file paths on the real filesystem. So instead I made a regex check. I am testing this locally, but the unit tests are passing.

File: tests/test_codeowners.py [hunk 1]
```diff
@@ -62,13 +62,13 @@ def test_pattern_match_single_component():
...
 
-@pytest.mark.xfail
 def test_pattern_match_recursive():
     rec_pat = codeowners.parse_pattern('a/**/b')
     assert rec_pat.match('a/b')
     assert rec_pat.match('a/x/b')
     assert rec_pat.match('a/x/y/z/b')
-    assert rec_pat.match('a/x/y/z/c')
+    # Does not end in 'b', no match.
+    assert not rec_pat.match('a/x/y/z/c')
```
Comment (shaleh) [inline tests/test_codeowners.py:71]: I think this was supposed to be a negated check correct? How could 'a/**/b' match a path without b?

Comment (nmusolino) [reply tests/test_codeowners.py:71]: Yes, thanks for fixing this.

**Pull request merged**